### PR TITLE
feat: add google sign in with oauth2

### PR DESCRIPTION
**Pull request by isavega**
# Base PR
- PR based on main

# Context
- With this PR, it is possible to login with google with omniauth gem

# Changelog

- Add new gem: omniaut for manage multi-provider authentication
- Add new gem: omniauth-google-oauth2 for authenticate with Google via OAuth2 in OmniAuth
- Add new gem: omniauth-rails_csrf_protection to handle csrf protection
- Create a new view for sign in
- Create a new view for sign up
- Create a new controller for omniauth callbacks
- Create a new controller for registrations
- Create a new controller for sessions
- Create a new method for user model
- Create migrations for add provider and oauth_id to user model

# QA
- Check if your email is on the list of Google Cloud testers
- Create .env file with the credentials
- Run: `docker compose build`
- Run: `docker compose run web rails db:reset`
- Run: `docker compose run web rails db:migrate`
- [FRONT] On the branch `feature/googleLogin` go to `http://localhost:3001/users/sign_in`

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -2503,13 +2503,13 @@ Metrics/CyclomaticComplexity:
   Max: 7
 
 Metrics/MethodLength:
-  Description: 'Avoid methods longer than 10 lines of code.'
+  Description: 'Avoid methods longer than 15 lines of code.'
```
Comment (AlehandroL) [inline .rubocop.yml:2506]: Nope

**Review by CRiquelme — APPROVED** [review]

**Comment by valentinjadot** [comment]
ke es esoooo
image

File: Gemfile.lock [hunk 2]
```diff
@@ -119,12 +119,20 @@ GEM
     diff-lcs (1.5.0)
     digest (3.1.0)
     docile (1.4.0)
+    dotenv (2.8.1)
```
Comment (vpulgar) [inline Gemfile.lock:122]: Is it necessary?

**Review by vpulgar — APPROVED** [review]

File: Gemfile.lock (hunk 2 above)
Comment (isavega) [reply Gemfile.lock:122]: Yes

**Review by AlehandroL — APPROVED** [review]

**Pull request merged**